internal/cephfs/volumemounter.go

```diff
@@ -289,16 +289,7 @@ func mountKernel(ctx context.Context, mountPoint string, cr *util.Credentials, v
 		mountPoint,
 	}
 	optionsStr := fmt.Sprintf("name=%s,secretfile=%s", cr.ID, cr.KeyFile)
-	if volOptions.FsName != "" {
-		optionsStr += fmt.Sprintf(",mds_namespace=%s", volOptions.FsName)
-	}
-	if volOptions.KernelMountOptions != "" {
-		optionsStr += fmt.Sprintf(",%s", volOptions.KernelMountOptions)
-	}
-
-	if !strings.Contains(volOptions.KernelMountOptions, netDev) {
-		optionsStr += fmt.Sprintf(",%s", netDev)
-	}
+	optionsStr = util.MountOptionsAdd(optionsStr, volOptions.FsName, volOptions.KernelMountOptions, netDev)
```

Comment:
`mds_namespace=%s"` is missing here.

Reply:
no, that is already set in `optionsStr` a line higher?

Reply:
nope, can you please check again
https://github.com/ceph/ceph-csi/blob/1aa106ff1d6bfc6f1b921e885d22b7aff9fb276f/internal/cephfs/volumemounter.go#L281-L297

Reply:
Ah, now I see the `mds_namespace=` piece is missing, the option is not correct. Thanks!

Reply:
this should have been addressed now too :facepalm: